wordpress-mobile/AztecEditor-iOS
# Issue/115 shortcodes reverse path

Refs #115 

This PR adds support for executing the step of converting HTML back to shortcodes.

The PR also refactor the original shortcode code in order to be more generic and share code with html processor.

To test:
 - Open the demo app
 - Run the demo with content
 - See if video show correctly
 - See if when converting back to html mode the videos are converted back to video shortcodes.
